src/python/pants/backend/python/tasks/resolve_requirements_task_base.py

```diff
@@ -62,13 +64,43 @@ def resolve_requirements(self, req_libs, local_dist_targets=None):
           # Handle locally-built python distribution dependencies.
           built_dists = self.context.products.get_data(BuildLocalPythonDistributions.PYTHON_DISTS)
           if built_dists:
-            req_libs = inject_synthetic_dist_requirements(self.context.build_graph,
-                                                          built_dists,
-                                                          ':'.join(2 * [target_set_id])) + req_libs
-          self._build_requirements_pex(interpreter, safe_path, req_libs)
+            req_libs = inject_synthetic_dist_requirements(
+              self.context.build_graph, built_dists, ':'.join(2 * [target_set_id])
+            ) + req_libs
+          builder = PEXBuilder(path=safe_path, interpreter=interpreter, copy=True)
+          dump_requirement_libs(builder, interpreter, req_libs, self.context.log)
+          builder.freeze()
     return PEX(path, interpreter=interpreter)
 
-  def _build_requirements_pex(self, interpreter, path, req_libs):
-    builder = PEXBuilder(path=path, interpreter=interpreter, copy=True)
-    dump_requirements(builder, interpreter, req_libs, self.context.log)
-    builder.freeze()
+  def resolve_requirement_strings(self, interpreter, requirement_strings):
+    """Resolve a list of pip-style requirement strings."""
+    requirement_strings = sorted(requirement_strings)
+    if len(requirement_strings) == 0:
+      req_strings_id = 'no_requirements'
+    elif len(requirement_strings) == 1:
+      req_strings_id = requirement_strings[0]
+    else:
+      req_strings_id = hash_all(requirement_strings)
+
+    path = os.path.realpath(os.path.join(self.workdir, str(interpreter.identity), req_strings_id))
+    if not os.path.isdir(path):
+      reqs = [PythonRequirement(req_str) for req_str in requirement_strings]
+      with safe_concurrent_creation(path) as safe_path:
+        builder = PEXBuilder(path=safe_path, interpreter=interpreter, copy=True)
+        dump_requirements(builder, interpreter, reqs, self.context.log)
+        builder.freeze()
+    return PEX(path, interpreter=interpreter)
+
+  @classmethod
+  def merge_pexes(cls, path, pex_info, interpreter, pexes, interpeter_constraints=None):
```

Comment:
Seems worth calling out `pex_info` is mutated or else doing a `pex_info.copy()`.

Reply:
Good idea, done (copied the pex_info).